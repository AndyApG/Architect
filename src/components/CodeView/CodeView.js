import React, { useMemo } from 'react';
import PropTypes from 'prop-types';
import { useSelector } from 'react-redux';
import cx from 'classnames';
import { motion } from 'framer-motion';
import { getFormValues } from 'redux-form';
import { getCSSVariableAsNumber } from '@codaco/ui/lib/utils/CSSVariables';

const variants = {
  hide: { opacity: 0, translateY: '-100%' },
  show: { opacity: 1, translateY: '0%' },
};

const CodeView = ({ toggleCodeView, show, form }) => {
  const code = useSelector(getFormValues(form));
  const transition = useMemo(() => ({
    ease: 'easeInOut',
    duration: getCSSVariableAsNumber('--animation-duration-standard-ms') / 1000,
  }), []);

  return (
    <motion.div
      className={cx('code-view', { 'code-view--show': show })}
      variants={variants}
      initial="hide"
      animate={show ? 'show' : 'hide'}
      transition={transition}
    >
      <div className="code-view__content">
        <pre>
          <code>
            { show && JSON.stringify(code, null, 2) }
          </code>
        </pre>
      </div>
      <div className="code-view__controls" onClick={toggleCodeView}>Close code view</div>
    </motion.div>
  );
};

CodeView.propTypes = {
  form: PropTypes.string.isRequired,
  toggleCodeView: PropTypes.func.isRequired,
  show: PropTypes.bool,
};

CodeView.defaultProps = {
  show: false,
};

export default CodeView;
