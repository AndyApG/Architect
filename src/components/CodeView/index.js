export { default as CodeView } from './CodeView';
export { default as FormCodeView } from './FormCodeView';
