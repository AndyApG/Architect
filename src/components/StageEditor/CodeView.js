import React from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import { getFormValues } from 'redux-form';

const CodeView = ({ toggleCodeView, stage }) => (
  <div className="stage-editor__code" onClick={toggleCodeView}>
    <pre>
      <code>
        { JSON.stringify(stage, null, 2) }
      </code>
    </pre>
  </div>
);

CodeView.propTypes = {
  stage: PropTypes.object,
  toggleCodeView: PropTypes.func.isRequired,
};

CodeView.defaultProps = {
  stage: {},
};

const mapStateToProps = (state, { form }) => ({
  stage: getFormValues(form.name)(state),
});

export default connect(mapStateToProps)(CodeView);
