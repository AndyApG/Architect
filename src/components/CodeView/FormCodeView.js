import { connect } from 'react-redux';
import { getFormValues } from 'redux-form';
import CodeView from './CodeView';

const mapStateToProps = (state, { form }) => ({
  code: getFormValues(form)(state),
});

export default connect(mapStateToProps)(CodeView);
