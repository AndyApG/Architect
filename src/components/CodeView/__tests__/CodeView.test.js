/* eslint-env jest */

import React from 'react';
import { shallow } from 'enzyme';
import CodeView from '../CodeView';

const mockProps = {
  toggleCodeView: () => {},
};

describe('<CodeView />', () => {
  it('can render', () => {
    const subject = shallow(<CodeView {...mockProps} />);
    expect(subject).toMatchSnapshot();
  });
});
