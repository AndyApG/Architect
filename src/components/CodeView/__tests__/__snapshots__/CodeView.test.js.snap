// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<CodeView /> can render 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <CodeView
    form="test"
    show={false}
    toggleCodeView={[Function]}
  >
    <ForwardRef(MotionComponent)
      animate="hide"
      className="code-view"
      initial="hide"
      transition={
        Object {
          "duration": NaN,
          "ease": "easeInOut",
        }
      }
      variants={
        Object {
          "hide": Object {
            "opacity": 0,
            "translateY": "-100%",
          },
          "show": Object {
            "opacity": 1,
            "translateY": "0%",
          },
        }
      }
    >
      <div
        className="code-view"
        style={
          Object {
            "opacity": 0,
            "transform": "translateY(-100%) translateZ(0)",
          }
        }
      >
        <div
          className="code-view__content"
        >
          <pre>
            <code />
          </pre>
        </div>
        <div
          className="code-view__controls"
          onClick={[Function]}
        >
          Close code view
        </div>
      </div>
      <Component
        animate="hide"
        className="code-view"
        inherit={false}
        initial="hide"
        key="animation"
        transition={
          Object {
            "duration": NaN,
            "ease": "easeInOut",
          }
        }
        variants={
          Object {
            "hide": Object {
              "opacity": 0,
              "translateY": "-100%",
            },
            "show": Object {
              "opacity": 1,
              "translateY": "0%",
            },
          }
        }
        visualElement={
          HTMLVisualElement {
            "activeOverrides": Set {},
            "axisProgress": Object {
              "x": MotionValue {
                "canTrackVelocity": true,
                "current": 0,
                "lastUpdated": 0,
                "prev": undefined,
                "renderSubscribers": SubscriptionManager {
                  "subscriptions": Set {},
                },
                "scheduleVelocityCheck": [Function],
                "timeDelta": 0,
                "updateAndNotify": [Function],
                "updateSubscribers": SubscriptionManager {
                  "subscriptions": Set {},
                },
                "velocityCheck": [Function],
              },
              "y": MotionValue {
                "canTrackVelocity": true,
                "current": 0,
                "lastUpdated": 0,
                "prev": undefined,
                "renderSubscribers": SubscriptionManager {
                  "subscriptions": Set {},
                },
                "scheduleVelocityCheck": [Function],
                "timeDelta": 0,
                "updateAndNotify": [Function],
                "updateSubscribers": SubscriptionManager {
                  "subscriptions": Set {},
                },
                "velocityCheck": [Function],
              },
            },
            "baseTarget": Object {
              "opacity": 0,
              "translateY": "-100%",
            },
            "children": Set {},
            "config": Object {
              "allowTransformNone": true,
              "animate": "hide",
              "children": Array [
                <div
                  className="code-view__content"
                >
                  <pre>
                    <code />
                  </pre>
                </div>,
                <div
                  className="code-view__controls"
                  onClick={[Function]}
                >
                  Close code view
                </div>,
              ],
              "className": "code-view",
              "enableHardwareAcceleration": true,
              "initial": "hide",
              "transition": Object {
                "duration": NaN,
                "ease": "easeInOut",
              },
              "variants": Object {
                "hide": Object {
                  "opacity": 0,
                  "translateY": "-100%",
                },
                "show": Object {
                  "opacity": 1,
                  "translateY": "0%",
                },
              },
            },
            "current": <div
              class="code-view"
              style="opacity: 0; transform: translateY(-100%) translateZ(0);"
            >
              <div
                class="code-view__content"
              >
                <pre>
                  <code />
                </pre>
              </div>
              <div
                class="code-view__controls"
              >
                Close code view
              </div>
            </div>,
            "defaultConfig": Object {
              "allowTransformNone": true,
              "enableHardwareAcceleration": true,
            },
            "delta": Object {
              "x": Object {
                "origin": 0,
                "originPoint": 0,
                "scale": 1,
                "translate": 0,
              },
              "y": Object {
                "origin": 0,
                "originPoint": 0,
                "scale": 1,
                "translate": 0,
              },
            },
            "deltaFinal": Object {
              "x": Object {
                "origin": 0,
                "originPoint": 0,
                "scale": 1,
                "translate": 0,
              },
              "y": Object {
                "origin": 0,
                "originPoint": 0,
                "scale": 1,
                "translate": 0,
              },
            },
            "deltaTransform": "translate3d(0px, 0px, 0) scale(1, 1)",
            "depth": 0,
            "element": <div
              class="code-view"
              style="opacity: 0; transform: translateY(-100%) translateZ(0);"
            >
              <div
                class="code-view__content"
              >
                <pre>
                  <code />
                </pre>
              </div>
              <div
                class="code-view__controls"
              >
                Close code view
              </div>
            </div>,
            "externalRef": null,
            "hasViewportBoxUpdated": false,
            "isAnimating": Set {},
            "isLayoutProjectionEnabled": false,
            "isMounted": true,
            "isPresenceRoot": true,
            "isPresent": true,
            "isTargetBoxLocked": false,
            "latest": Object {
              "opacity": 0,
              "translateY": "-100%",
            },
            "layoutId": undefined,
            "layoutMeasureListeners": SubscriptionManager {
              "subscriptions": Set {},
            },
            "layoutUpdateListeners": SubscriptionManager {
              "subscriptions": Set {},
            },
            "overrides": Array [],
            "parent": undefined,
            "prevTreeScale": Object {
              "x": 1,
              "y": 1,
            },
            "reactStyle": Object {},
            "ref": [Function],
            "resolvedOverrides": Array [],
            "rootParent": [Circular],
            "stopLayoutAxisAnimation": Object {
              "x": [Function],
              "y": [Function],
            },
            "style": Object {
              "opacity": 0,
              "transform": "translateY(-100%) translateZ(0)",
            },
            "targetBoxFinal": Object {
              "x": Object {
                "max": 1,
                "min": 0,
              },
              "y": Object {
                "max": 1,
                "min": 0,
              },
            },
            "transform": Object {
              "translateY": "-100%",
            },
            "transformKeys": Array [
              "translateY",
            ],
            "transformOrigin": Object {},
            "treePath": Array [],
            "treeScale": Object {
              "x": 1,
              "y": 1,
            },
            "triggerRender": [Function],
            "update": [Function],
            "updateLayoutDelta": [Function],
            "valueSubscriptions": Map {
              "opacity" => [Function],
              "translateY" => [Function],
            },
            "values": Map {
              "opacity" => MotionValue {
                "canTrackVelocity": true,
                "current": 0,
                "lastUpdated": 0,
                "prev": undefined,
                "renderSubscribers": SubscriptionManager {
                  "subscriptions": Set {
                    [Function],
                  },
                },
                "scheduleVelocityCheck": [Function],
                "timeDelta": 0,
                "updateAndNotify": [Function],
                "updateSubscribers": SubscriptionManager {
                  "subscriptions": Set {
                    [Function],
                  },
                },
                "velocityCheck": [Function],
              },
              "translateY" => MotionValue {
                "canTrackVelocity": true,
                "current": "-100%",
                "lastUpdated": 0,
                "prev": undefined,
                "renderSubscribers": SubscriptionManager {
                  "subscriptions": Set {
                    [Function],
                  },
                },
                "scheduleVelocityCheck": [Function],
                "timeDelta": 0,
                "updateAndNotify": [Function],
                "updateSubscribers": SubscriptionManager {
                  "subscriptions": Set {
                    [Function],
                  },
                },
                "velocityCheck": [Function],
              },
            },
            "vars": Object {},
            "viewportBoxUpdateListeners": SubscriptionManager {
              "subscriptions": Set {},
            },
          }
        }
      />
    </ForwardRef(MotionComponent)>
  </CodeView>
</Provider>
`;

exports[`<CodeView /> renders content only when show is true 1`] = `
<Provider
  store={
    Object {
      "dispatch": [Function],
      "getState": [Function],
      "replaceReducer": [Function],
      "subscribe": [Function],
      Symbol(observable): [Function],
    }
  }
>
  <CodeView
    form="test"
    show={true}
    toggleCodeView={[Function]}
  >
    <ForwardRef(MotionComponent)
      animate="show"
      className="code-view code-view--show"
      initial="hide"
      transition={
        Object {
          "duration": NaN,
          "ease": "easeInOut",
        }
      }
      variants={
        Object {
          "hide": Object {
            "opacity": 0,
            "translateY": "-100%",
          },
          "show": Object {
            "opacity": 1,
            "translateY": "0%",
          },
        }
      }
    >
      <div
        className="code-view code-view--show"
        style={
          Object {
            "opacity": 0,
            "transform": "translateY(-100%) translateZ(0)",
          }
        }
      >
        <div
          className="code-view__content"
        >
          <pre>
            <code>
              {
  "name": "example name"
}
            </code>
          </pre>
        </div>
        <div
          className="code-view__controls"
          onClick={[Function]}
        >
          Close code view
        </div>
      </div>
      <Component
        animate="show"
        className="code-view code-view--show"
        inherit={false}
        initial="hide"
        key="animation"
        transition={
          Object {
            "duration": NaN,
            "ease": "easeInOut",
          }
        }
        variants={
          Object {
            "hide": Object {
              "opacity": 0,
              "translateY": "-100%",
            },
            "show": Object {
              "opacity": 1,
              "translateY": "0%",
            },
          }
        }
        visualElement={
          HTMLVisualElement {
            "activeOverrides": Set {},
            "axisProgress": Object {
              "x": MotionValue {
                "canTrackVelocity": true,
                "current": 0,
                "lastUpdated": 0,
                "prev": undefined,
                "renderSubscribers": SubscriptionManager {
                  "subscriptions": Set {},
                },
                "scheduleVelocityCheck": [Function],
                "timeDelta": 0,
                "updateAndNotify": [Function],
                "updateSubscribers": SubscriptionManager {
                  "subscriptions": Set {},
                },
                "velocityCheck": [Function],
              },
              "y": MotionValue {
                "canTrackVelocity": true,
                "current": 0,
                "lastUpdated": 0,
                "prev": undefined,
                "renderSubscribers": SubscriptionManager {
                  "subscriptions": Set {},
                },
                "scheduleVelocityCheck": [Function],
                "timeDelta": 0,
                "updateAndNotify": [Function],
                "updateSubscribers": SubscriptionManager {
                  "subscriptions": Set {},
                },
                "velocityCheck": [Function],
              },
            },
            "baseTarget": Object {
              "opacity": 1,
              "translateY": "0%",
            },
            "children": Set {},
            "config": Object {
              "allowTransformNone": true,
              "animate": "show",
              "children": Array [
                <div
                  className="code-view__content"
                >
                  <pre>
                    <code>
                      {
  "name": "example name"
}
                    </code>
                  </pre>
                </div>,
                <div
                  className="code-view__controls"
                  onClick={[Function]}
                >
                  Close code view
                </div>,
              ],
              "className": "code-view code-view--show",
              "enableHardwareAcceleration": true,
              "initial": "hide",
              "transition": Object {
                "duration": NaN,
                "ease": "easeInOut",
              },
              "variants": Object {
                "hide": Object {
                  "opacity": 0,
                  "translateY": "-100%",
                },
                "show": Object {
                  "opacity": 1,
                  "translateY": "0%",
                },
              },
            },
            "current": <div
              class="code-view code-view--show"
              style="opacity: 0; transform: translateY(-100%) translateZ(0);"
            >
              <div
                class="code-view__content"
              >
                <pre>
                  <code>
                    {
  "name": "example name"
}
                  </code>
                </pre>
              </div>
              <div
                class="code-view__controls"
              >
                Close code view
              </div>
            </div>,
            "defaultConfig": Object {
              "allowTransformNone": true,
              "enableHardwareAcceleration": true,
            },
            "delta": Object {
              "x": Object {
                "origin": 0,
                "originPoint": 0,
                "scale": 1,
                "translate": 0,
              },
              "y": Object {
                "origin": 0,
                "originPoint": 0,
                "scale": 1,
                "translate": 0,
              },
            },
            "deltaFinal": Object {
              "x": Object {
                "origin": 0,
                "originPoint": 0,
                "scale": 1,
                "translate": 0,
              },
              "y": Object {
                "origin": 0,
                "originPoint": 0,
                "scale": 1,
                "translate": 0,
              },
            },
            "deltaTransform": "translate3d(0px, 0px, 0) scale(1, 1)",
            "depth": 0,
            "element": <div
              class="code-view code-view--show"
              style="opacity: 0; transform: translateY(-100%) translateZ(0);"
            >
              <div
                class="code-view__content"
              >
                <pre>
                  <code>
                    {
  "name": "example name"
}
                  </code>
                </pre>
              </div>
              <div
                class="code-view__controls"
              >
                Close code view
              </div>
            </div>,
            "externalRef": null,
            "hasViewportBoxUpdated": false,
            "isAnimating": Set {
              "opacity",
              "translateY",
            },
            "isLayoutProjectionEnabled": false,
            "isMounted": true,
            "isPresenceRoot": true,
            "isPresent": true,
            "isTargetBoxLocked": false,
            "latest": Object {
              "opacity": 0,
              "translateY": "-100%",
            },
            "layoutId": undefined,
            "layoutMeasureListeners": SubscriptionManager {
              "subscriptions": Set {},
            },
            "layoutUpdateListeners": SubscriptionManager {
              "subscriptions": Set {},
            },
            "overrides": Array [],
            "parent": undefined,
            "prevTreeScale": Object {
              "x": 1,
              "y": 1,
            },
            "reactStyle": Object {},
            "ref": [Function],
            "resolvedOverrides": Array [],
            "rootParent": [Circular],
            "stopLayoutAxisAnimation": Object {
              "x": [Function],
              "y": [Function],
            },
            "style": Object {
              "opacity": 0,
              "transform": "translateY(-100%) translateZ(0)",
            },
            "targetBoxFinal": Object {
              "x": Object {
                "max": 1,
                "min": 0,
              },
              "y": Object {
                "max": 1,
                "min": 0,
              },
            },
            "transform": Object {
              "translateY": "-100%",
            },
            "transformKeys": Array [
              "translateY",
            ],
            "transformOrigin": Object {},
            "treePath": Array [],
            "treeScale": Object {
              "x": 1,
              "y": 1,
            },
            "triggerRender": [Function],
            "update": [Function],
            "updateLayoutDelta": [Function],
            "valueSubscriptions": Map {
              "opacity" => [Function],
              "translateY" => [Function],
            },
            "values": Map {
              "opacity" => MotionValue {
                "canTrackVelocity": true,
                "current": 0,
                "lastUpdated": 0,
                "prev": undefined,
                "renderSubscribers": SubscriptionManager {
                  "subscriptions": Set {
                    [Function],
                  },
                },
                "scheduleVelocityCheck": [Function],
                "stopAnimation": [Function],
                "timeDelta": 0,
                "updateAndNotify": [Function],
                "updateSubscribers": SubscriptionManager {
                  "subscriptions": Set {
                    [Function],
                  },
                },
                "velocityCheck": [Function],
              },
              "translateY" => MotionValue {
                "canTrackVelocity": true,
                "current": "-100%",
                "lastUpdated": 0,
                "prev": undefined,
                "renderSubscribers": SubscriptionManager {
                  "subscriptions": Set {
                    [Function],
                  },
                },
                "scheduleVelocityCheck": [Function],
                "stopAnimation": [Function],
                "timeDelta": 0,
                "updateAndNotify": [Function],
                "updateSubscribers": SubscriptionManager {
                  "subscriptions": Set {
                    [Function],
                  },
                },
                "velocityCheck": [Function],
              },
            },
            "vars": Object {},
            "viewportBoxUpdateListeners": SubscriptionManager {
              "subscriptions": Set {},
            },
          }
        }
      />
    </ForwardRef(MotionComponent)>
  </CodeView>
</Provider>
`;
